src/RpcServer/RelayActor.cs

```diff
@@ -0,0 +1,40 @@
+using Akka.Actor;
+using Neo.Network.P2P.Payloads;
+using static Neo.Ledger.Blockchain;
+
+namespace Neo.Plugins
+{
+    public class RelayActor : UntypedActor
+    {
+        private readonly IActorRef blockchain;
+        private IInventory inventory;
+        private IActorRef sender;
+
+        public RelayActor(IActorRef blockchain)
+        {
+            this.blockchain = blockchain;
+            Context.System.EventStream.Subscribe(Self, typeof(RelayResult));
+        }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case IInventory inventory:
+                    this.inventory = inventory;
+                    this.sender = Sender;
+                    blockchain.Tell(inventory);
+                    break;
+                case RelayResult reason:
+                    if (reason.Inventory.Hash.Equals(inventory.Hash))
```

Comment:
The inventory may change when another request comes before the first return. Maybe we can maintain a `Dictionary<UInt256, RelayResult>` to avoid this error.

Reply:
You are right! There should be a dictionary.